### DISCUSSION TITLE
Welcome to AI-SBC Discussions!

### DISCUSSION THREAD

**Discussion by titiroMonkey** [ThomasSteere/AI-SBC#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ShahabSotouni** [comment]
I haven't tried the repository, but loved the idea. Good Job!

**Comment by roxnestriker** [comment]
Fantastic work! The update is absolutely mind-blowing — everything is working smoothly.

Just wanted to suggest one dream feature:
It would be great to have an "Ignore Max Player Rating for TOTW/TOTS" toggle or checkbox.

For example, if I set the Max Player Rating to 90, it should still allow 95-rated TOTW or TOTS players to be used (ignoring the rating cap just for special cards). This would offer more flexibility when building squads.

Thanks again for the amazing work — truly impressive!

**Comment by sam00787** [comment]
Hello when  i try to solve prime silver sbc it uses gold players
<img width="1242" height="519" alt="Screenshot from 2025-11-09 20-05-39" src="https://github.com/user-attachments/assets/6c64303d-6559-48ac-aaa1-d960029bab8b" />

**Reply by titiroMonkey** [reply]
You sbc storage is discounted by x% determined by the settings. Default is 50% so a 700 coin player in the solve is 350

**Reply by sam00787** [reply]
ok thank you i will check

**Reply by titiroMonkey** [reply]
It creates a file called allplayers.csv and there is a column called price which is a calculated price used to solve. 
Check out the settings page and adjust these to your liking. I like to add max ratings and add in exclude filters

**Comment by sam00787** [comment]
can it solve league sbc?

**Reply by titiroMonkey** [reply]
It can solve all sbcs. If you are using concepts i would recommend to increase the api max time to allow for more players

**Reply by sam00787** [reply]
ok thank you

**Reply by sam00787** [reply]
it solved all league sbc but is having trouble with the librdatores one i put concept on

**Reply by titiroMonkey** [reply]
I put max solve to 999 for that one

**Reply by sam00787** [reply]
do i need to change any other setting for concepts

**Reply by sam00787** [reply]
I increased max time still can't use concept

**Reply by titiroMonkey** [reply]
Im going to add a special case solve for the league sbcs as they seem to be the ones everyone wants to solve with concepts. Keep an eye out for updates

**Comment by roxnestriker** [comment]
<img width="2172" height="1078" alt="image" src="https://github.com/user-attachments/assets/18bb5b5c-8a3d-4ec3-a70b-2ef7e15f9f78" />
Please see the solution provided by the script above.

As shown in the second screenshot, the script is not selecting the cheapest players for the solution.
<img width="2192" height="1061" alt="image" src="https://github.com/user-attachments/assets/f72228f9-fc70-42b8-bb1e-16138cf268d3" />

Additionally, I am unable to find an important setting: how to exclude players from my active team when completing an SBC.

**Reply by titiroMonkey** [reply]
Are they storage players?
 Storage players value is less than others based on the settings tab. The solution is based on a players value which takes into account a storage or duplicate discount. Also a rating discount.
If you want to purely use price you need to adjust the discount sliders accordingly.

You can also exclude players in the settings tab as well or by selecting them and clicking the sbc lock button.

**Reply by roxnestriker** [reply]
All of these are club players; I just started a fresh account and don't have anything in storage.

I believe there is an option in the EA FC 25 script to exclude squad players. This is a very useful feature since we change our squads frequently. Additionally, while the script correctly considers player prices, it should prioritize using lower-rated players when prices are equal. For example, if a 78-rated and an 82-rated player cost the same, it should use the 78-rated one so the 82-rated player can be saved for higher-requirement SBCs.

These are just my suggestions. Overall, this is a great script and likely the best tool available. Thank you for such an excellent tool!

**Reply by titiroMonkey** [reply]
Heres an ai explanation of how sbc price is found. Step 6 takes rating into account so a card with the same price has a lower value the lower rated it is.


The function responsible for determining the "sbcprice" in your project is called getSBCPrice. Here’s how it works, step-by-step:

1. **Checks if the item is fixed**  
   - If the item is fixed (using isItemFixed), sbcprice is set to 1.

2. **Base price calculation**  
   - It calculates an initial price (`sbcPrice`) as the maximum of:
     - getPrice(item)
     - getPrice({ definitionId: item.rating + "_CBR" })
     - 100

3. **Price not found scenario**  
   - If getPrice(item) returns -1 (suggesting the price was not found), the function returns 1.5 times the calculated sbcPrice.

4. **Concept item price adjustment**  
   - If the item is a concept item (`item.concept`), it multiplies sbcPrice by a user-defined conceptPremium value from settings.

5. **Evolution card price adjustment**  
   - If the item matches certain card types (detected via string includes "volution"), it multiplies sbcPrice by a user-defined evoPremium value from settings.

6. **Rating-based discount**  
   - The function reduces sbcPrice by (100 - item.rating) as a discount for higher-rated items.

7. **Duplicate and storage discounts**  
   - If the item is a duplicate or is in storage, sbcPrice is multiplied by a duplicateDiscount percentage (from settings).

8. **Untradeable discount**  
   - If the item isn’t tradeable, sbcPrice is multiplied by an untradeableDiscount percentage (also from settings).

9. **Returns the final sbcPrice**  
   - The function finally returns the computed sbcPrice.

**Code location:**  
You can view the function here in your repo:  
[plainJavascript.js#L5721-L5828](https://github.com/titiroMonkey/Auto-SBC/blob/main/plainJavascript.js#L5721-L5828)
@

**Reply by Regista6** [reply]
This is very neat. Another way to deal with this would be to run the optimizer twice. In the 2nd run, the objective would be to minimize the squad rating or minimize max rating while keeping the cost same as what was found during the 1st run.
